File: dsa-in-c&&cpp/bubbleSort.c
#include <stdio.h>

int bubbleSort(int Arr[],int length){
    for (int i = 0; i < length; i++)
    {
        for (int j = 0; j < length - 1; j++)
        {
            int copy;
            if (Arr[j] > Arr[j + 1])
            {
                copy = Arr[j];
                Arr[j] = Arr[j + 1];
                Arr[j + 1] = copy;
            }
        }
    }

}

int main()
{
    int length;
    printf("Enter length of array : ");
    scanf("%d", &length);
    int Arr[length];
    printf("Enter values of array \n");
    for (int i = 0; i < length; i++)
    {
        printf("Enter value %d : ", i+1 );
        scanf("%d", &Arr[i]);
    }


    int result = bubbleSort(Arr, length);
   

    printf("Arr[%d]= [", length);
    for (int i = 0; i < length; i++)
    {

        printf(" %d ", Arr[i]);

        if (i < length - 1)
        {
            printf(",");
        }
        else
        {
            printf("]");
        }
    }

    return 0;
}

File: 04_Sorting/bubble.sort.js
let arr = [5,4,3,2,1]

let len = arr.length

for(let p = 0 ; p < arr.length - 1 ; p++){
    for (let j = 0; j < len - 1 - p ; j++) {
        if(arr[j] > arr[j+1]){
            let temp = arr[j]
            arr[j] = arr[j+1]
            arr[j+1] = temp
        }
        
    }
}

console.log(arr);